### PR TITLE
Add support for 32bit CI build

### PR DESCRIPTION
**Pull request by Krishna-13-cyber**
Addresses #748

### CODE REVIEW

**Comment by vgvassilev** [comment]
Does that work for PRs? I don’t see a build.

**Comment by Krishna-13-cyber** [comment]
> Does that work for PRs? I don’t see a build.

No then, I have enabled it now.

File: .github/workflows/platforms.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+name: 'Platforms'
```
Comment (vgvassilev) [inline .github/workflows/platforms.yml:2]:
```suggestion
name: 'Arch'
```

File: .github/workflows/platforms.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+           - name: "Setup"
+             run: |
+               clang --version
+               make --version
+             shell: alpine.sh {0}
+
+           - name: "Compile library"
+             run: |
+               mkdir build && cd build
+               cmake -DLLVM_DIR=/usr/lib/ -DCMAKE_BUILD_TYPE=DEBUG -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
```
Comment (vgvassilev) [inline .github/workflows/platforms.yml:43]:
```suggestion
               cmake -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
```

File: .github/workflows/platforms.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
```
Comment (vgvassilev) [inline .github/workflows/platforms.yml:79]:
```suggestion
               cmake -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
```

File: .github/workflows/platforms.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
```
Comment (vgvassilev) [inline .github/workflows/platforms.yml:115]:
```suggestion
               cmake -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
```

Can we extract the step "Compile library" into one place as it seems to repeat across builds. The same for the other common steps. Maybe [this](https://stackoverflow.com/questions/59757355/reuse-portion-of-github-action-across-jobs) could help?

Comment (Krishna-13-cyber) [reply .github/workflows/platforms.yml:115]: Sorry for the delayed response. Yes, I will look into this and patch it asap.

Comment (vgvassilev) [reply .github/workflows/platforms.yml]: Can you take a look how the clad main workflow is organized, we want something similar for this PR.

Comment (Krishna-13-cyber) [reply .github/workflows/platforms.yml]: Yes, will do this.
Thanks!

Comment (Krishna-13-cyber) [reply .github/workflows/platforms.yml]: Hi, I have updated with respect to clad convention adhering to matrix strategy.
Thanks!

**Comment by vgvassilev** [comment]
Nice! Now we need to figure out how to suppress the test failures. For this PR maybe it would be fine if we allow failures on armv7 and x86.

File: .github/workflows/arch.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  architecture:
+    strategy:
+      matrix:
+        target: [x86, armv7, aarch64]
```
Comment (vgvassilev) [inline .github/workflows/arch.yml:15]: Looks like `armv7` and `aarch64` are considerable slower. Can we disable them?

**Comment by Krishna-13-cyber** [comment]
> Nice! Now we need to figure out how to suppress the test failures. For this PR maybe it would be fine if we allow failures on armv7 and x86.

Yes, we can solve the failures in a seperate PR.

**Comment by vgvassilev** [comment]
> > Nice! Now we need to figure out how to suppress the test failures. For this PR maybe it would be fine if we allow failures on armv7 and x86.
> 
> Yes, we can solve the failures in a seperate PR.

Looks like this won't work. Maybe in this PR we should add to each failing test: `XFAIL: target={{(i686|i386).*}}`. Can you try this and then we can drop the non-functional `continue-on-error` thing.

File: .github/workflows/arch.yml (hunk 5 above)
Comment (Krishna-13-cyber) [reply .github/workflows/arch.yml:15]: Yes, they are slower.
We had a intention to target multiple architectures right to get hold of failures/bugs so why should we disable them (due to time constraints).
Can we try any other workaround to fit in those(slower ones) as well?

**Comment by Krishna-13-cyber** [comment]
> > > Nice! Now we need to figure out how to suppress the test failures. For this PR maybe it would be fine if we allow failures on armv7 and x86.
> > 
> > 
> > Yes, we can solve the failures in a seperate PR.
> 
> Looks like this won't work. Maybe in this PR we should add to each failing test: `XFAIL: target={{(i686|i386).*}}`. Can you try this and then we can drop the non-functional `continue-on-error` thing.


I added this `continue-on-error` for all the parallel jobs to run independently as without this thing other jobs abort due to any failure on any of the jobs present.
Yes, I will try adding this `XFAIL: target={{(i686|i386).*}}` to failing tests.

**Comment by Krishna-13-cyber** [comment]
> Looks like this won't work. Maybe in this PR we should add to each failing test: `XFAIL: target={{(i686|i386).*}}`. Can you try this and then we can drop the non-functional `continue-on-error` thing.

The failing tests get suppressed with ```XFAIL:*``` and not this `XFAIL: target={{(i686|i386).*}}`.
Should I use this?

**Comment by vgvassilev** [comment]
> > Looks like this won't work. Maybe in this PR we should add to each failing test: `XFAIL: target={{(i686|i386).*}}`. Can you try this and then we can drop the non-functional `continue-on-error` thing.
> 
> The failing tests get suppressed with `XFAIL:*` and not this `XFAIL: target={{(i686|i386).*}}`. Should I use this?

No, that will suppress even working tests.

**Comment by vgvassilev** [comment]
> target={{(i686|i386).*}}

@Krishna-13-cyber, you need `// XFAIL: target={{i586.*}}`

**Review by vgvassilev — APPROVED** [review]
Awesome! LGTM!

**Pull request merged**